src/character.cpp

```diff
@@ -1061,9 +1066,38 @@ void Character::set_healthy_mod(int nhealthy_mod)
 {
     healthy_mod = nhealthy_mod;
 }
-void Character::mod_healthy_mod(int nhealthy_mod)
+void Character::mod_healthy_mod(int nhealthy_mod, int cap)
 {
+    // TODO: This really should be a full morale-like system, with per-effect caps
+    //       and durations.  This version prevents any single effect from exceeding its
+    //       intended ceiling, but multiple effects will overlap instead of adding.
+
+    // Cap indicates how far the mod is allowed to shift in this direction.
+    // It can have a different sign to the mod, e.g. for items that treat
+    // extremely low health, but can't make you healthy.
+    int low_cap, high_cap;
+    if (nhealthy_mod < 0) {
+        low_cap = cap;
+        high_cap = 200;
+    } else {
+        low_cap = -200;
+        high_cap = cap;
+    }
+
+    // If we're already out-of-bounds, we don't need to do anything.
+    if (healthy_mod <= low_cap || healthy_mod >= high_cap) {
```

Comment:
This will work in most cases, but will break if it gets out of bounds for some reason.
Changing it to
```
(healthy_mod <= low_cap && nhealthy_mod < 0) || (healthy_mod >= high_cap && nhealthy_mod > 0)
```

Would make it a bit more bug-resistant.